src/app/modules/workspace/__tests__/workspace.service.spec.ts

```diff
@@ -275,4 +278,181 @@ describe('workspace.service', () => {
       expect(actual._unsafeUnwrapErr()).toBeInstanceOf(WorkspaceNotFoundError)
     })
   })
+
+  describe('deleteWorkspace', () => {
+    const mockFormId = new ObjectId()
+    const mockWorkspace = {
+      _id: new ObjectId(),
+      admin: new ObjectId(),
+      title: 'workspace1',
+      formIds: [mockFormId],
+      count: 0,
+    }
+
+    it('should return undefined when successfully deleted workspace', async () => {
+      jest.spyOn(WorkspaceModel, 'deleteWorkspace').mockResolvedValueOnce(true)
+
+      const actual = await WorkspaceService.deleteWorkspace({
+        workspaceId: mockWorkspace._id.toHexString(),
+        shouldDeleteForms: false,
+      })
+
+      expect(actual.isOk()).toEqual(true)
+      expect(actual._unsafeUnwrap()).toEqual(undefined)
+    })
+
+    it('should return undefined when failed to delete workspace', async () => {
+      jest.spyOn(WorkspaceModel, 'deleteWorkspace').mockResolvedValueOnce(false)
+
+      const actual = await WorkspaceService.deleteWorkspace({
+        workspaceId: mockWorkspace._id.toHexString(),
+        shouldDeleteForms: false,
+      })
+
+      expect(actual.isOk()).toEqual(true)
+      expect(actual._unsafeUnwrap()).toEqual(undefined)
+    })
+
+    it('should not archive forms in the workspace when shouldDeleteForms is false', async () => {
+      await dbHandler.insertEncryptForm({
+        formId: mockFormId,
+        userId: mockWorkspace.admin,
+      })
+      const createdWorkspace = await WorkspaceModel.create(mockWorkspace)
+
+      jest
+        .spyOn(WorkspaceModel, 'findOne')
+        .mockResolvedValueOnce(createdWorkspace)
+      jest.spyOn(WorkspaceModel, 'deleteWorkspace').mockResolvedValueOnce(true)
+
+      const actual = await WorkspaceService.deleteWorkspace({
+        workspaceId: mockWorkspace._id.toHexString(),
+        shouldDeleteForms: false,
+      })
+      const doesFormExist = await FormModel.exists({ _id: mockFormId })
+      const isFormArchived = await FormModel.exists({
+        _id: mockFormId,
+        status: FormStatus.Archived,
+      })
```

Comment:
instead of making two async calls, we can just retrieve the form once and check that it is not null and that status is not archived

Reply:
Updated to do this instead! Thanks